### DISCUSSION TITLE
Default value for number field

### DISCUSSION THREAD

**Discussion by thomaskamalakis** [Budibase/budibase#6272] [Help] [answered]
OK so just a basic question: How can I set the default value of a number field equal to 1?

using:

- "1"
- 1
- {{ 1 }}

does not work. What works however is something much more complex like:

```
{{ date now "MM" }}
```

Default values seem to work great with text fields. What's happening here?

BR

Thomas

**Comment by melohagan** [comment, marked as answer]
Hi @thomaskamalakis 

Handlebar data types are always strings. If you want a number type, then you need to use a JavaScript binding e.g. `return 1`

**Reply by thomaskamalakis** [reply]
Many thanks

**Reply by thomaskamalakis** [reply]
@melohagan 
It seems that you can not set a default value equal to zero

So `return 1` works but `return 0` just leaves an empty value on the form field.

**Reply by melohagan** [reply]
Must be because zero is falsey. I've raised a bug here: https://github.com/Budibase/budibase/issues/6284
